pkg/sql/tenant.go

```diff
@@ -698,3 +698,33 @@ WHERE id = $1`, tenantID, tenantName); err != nil {
 
 	return nil
 }
+
+// GetTenantInfo implements the tree.TenantOperator interface.
+func (p *planner) GetTenantInfo(
+	ctx context.Context, tenantName roachpb.TenantName,
+) (*descpb.TenantInfo, error) {
```

Comment:
[nit] It's not particularly a lot of code so I don't mind the duplication too much, but this function is pretty much GetTenantRecordByName with an additional role check.  Perhaps we can call the other function?